lib/init.coffee

```diff
@@ -1,13 +1,68 @@
+{BufferedProcess, CompositeDisposable} = require 'atom'
+
 module.exports =
   config:
     puppetLintExecutablePath:
-      default: ''
+      default: 'puppet-lint'
       title: 'Puppet Lint Executable Path'
       type: 'string'
     puppetLintArguments:
-      default: '--no-autoloader_layout-check'
+      default: [ '--no-autoloader_layout-check' ]
       title: 'Puppet Lint Arguments'
       type: 'string'
 
   activate: ->
-    console.log 'activate linter-puppet-lint'
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-puppet-lint.puppetLintExecutablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-puppet-lint.puppetLintArguments',
+      (args) =>
+        @args = args.split(' ')
+  deactivate: ->
+    @subscriptions.dispose()
+
+  puppetLinter: ->
+    provider =
+      grammarScopes: ['source.puppet']
+      scope: 'file'
+      lintOnFly: true
+      lint: (textEditor) =>
+        return new Promise (resolve, reject) =>
+          filePath = textEditor.getPath()
+          regex = /(warning|error): (.+?) on line (\d+) col (\d+)/
+          msg = ''
+          arg = @args.slice(0)
+          arg.push "--log-format"
+          arg.push "'%{kind}: %{message} on line %{line} col %{column}'"
+          arg.push filePath
+          process = new BufferedProcess
```

Comment:
If this is a requirement I'll rework it to use the atom-linter helper API.  If I had known about that from the get-go I would of used it off the bat... Not sure if it's worth the effort to convert to for 4 -- operations though.